### ISSUE TITLE
What is going on with Regions?

### ISSUE THREAD

**Issue by macasas** [tamas-p/hisma#23]
I am having problems with Regions, fine until I add a entryConnection or exitConnection. I thought it was my code for ages, but then ran the example, and it does the same thing.

Here is the given example 04_entry_and_exit_points with just the main and power machine, not linked to 02 and 03 for colour and brightness.  I cannot fit it all on a screen and still see details.

It's as if the regions are going in a recursive loop of some kind.  

Any ideas what is causing this?

<img width="1512" alt="Screenshot 2023-12-25 at 23 43 12" src="https://github.com/tamas-p/hisma/assets/910082/77ff28ae-25be-4421-b2e6-aeb0bb815bc4">

<img width="1378" alt="Screenshot 2023-12-25 at 23 51 29" src="https://github.com/tamas-p/hisma/assets/910082/3b126f36-6cb0-4760-86c1-19a2df6b25fd">

**Comment by tamas-p** [comment]
I am trying to reproduce the problem.
If I execute the command
```console
$ dart packages/hisma/example/04_entry_and_exit_points.dart
```
I get the expected behavior:

https://github.com/tamas-p/hisma/assets/10595775/8227ff0d-3150-46ee-abff-e464e8553796

What code changes shall I do to reproduce the issue?
You might also put in your reply the diff for the file.

**Comment by macasas** [comment]
Hi, the differences I see are you running dart, I am running flutter, but the code is the same.

As I say in the videos, it seems to be in the on state, and caused by the inclusion of entry/exit connectors, because taking them out and the problem is not there.

Couldn't upload videos here, too large, so here are 2 links to them on youtube.

What am I doing wrong?  Thanks

Ex 1 cut down version
[https://youtu.be/3E-8quupFAY](https://youtu.be/3E-8quupFAY)

Ex 2, all regions
[https://youtu.be/P_Sz0lwGMGI](https://youtu.be/P_Sz0lwGMGI)

**Comment by tamas-p** [comment]
These videos are private. Please change them from Private to Unlisted.

**Comment by macasas** [comment]
Sorry, i’ve done that.

**Comment by tamas-p** [comment]
Links work now. I will repeat your steps later today.

**Comment by tamas-p** [comment]
Hello, the good news that I found out what caused that strange visma behavior: It seems PlantUML had some changes recently that introduced non backward compatible changes either in its syntax or their rendering or both. I noticed from your video that you used plantuml-1.2023.13.jar, so I downloaded this (today the latest) version and it created the very same bad rendering for me as it did for you. The quick fix is sticking with an older version of PlantUML: https://sourceforge.net/projects/plantuml/files/1.2022.1/ (this is also the version used in the visma docker image).
I have already noticed rendering bugs in newer PlantUML versions back in 2022 (e.g. https://github.com/plantuml/plantuml/issues/1241 and https://github.com/plantuml/plantuml/issues/1269), that was why I recommended staying with PlantUML [V1.2022.1](https://sourceforge.net/projects/plantuml/files/1.2022.1/). Now based on this new finding I will highlight this need even more.

Besides this non-backward compatibility I have one more concern regarding PlantUML that it needs a JVM. Another rendering engine I consider adding is https://mermaid.js.org/.

**Comment by macasas** [comment]
Oh sorry, I must have missed your plantuml version recommendation when I was confused last time trying to connect to visma.  Still trying to get my head around it all.  The problem now makes perfect sense, thank you.

**Comment by tamas-p** [comment]
It was caused by a breaking change in the PlantUML API in plantuml-1.2023.2 as described in https://github.com/plantuml/plantuml/issues/1648
I committed the fix to master: 50e31cbc9ee8a9f57d4b5e48f874ec441761cfdc

**Comment by tamas-p** [comment]
Released in https://pub.dev/packages/hisma/versions/0.3.2+1
